### ISSUE TITLE
(0.9.0) /settings doesn't return 403 Forbidden

### ISSUE THREAD

**Issue by peterkaminski** [GovReady/govready-q#750]
This may be an important bug, because it represents a fault in authorization.

I observed this in testing, as: `ERROR: test_settings_page (siteapp.tests.OrganizationSettingsTests)`

The particular test is:

```
response = self.client.get('/settings')
self.assertEqual(response.status_code, 403)
```

It appears that `/settings` does not return a `403 Forbidden` status for a user who doesn't have organizational permission.  Rather, it allows access to `/settings` (presumably on its way to returning a `200 OK` status), but then hits an exception related to not having an organization.

Further, the test does not fail properly, because the exception is hit before HTTP status is returned, so the error is not detected nor logged properly.

Test set up and run:

```
git checkout 0.9.0.dev
git pull
source env/bin/activate
pip install -r requirements.txt
rm local/db.sqlite3
python3 manage.py migrate
python3 manage.py load_modules
 python3 manage.py first_run
./fetch-vendor-resources.sh
python3 manage.py test
```

Output:

```
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
..................login ip=127.0.0.1 username=me result=user:2
.login ip=127.0.0.1 username=me result=user:4
..login ip=127.0.0.1 username=me result=user:13
.login ip=127.0.0.1 username=me result=user:17
..login ip=127.0.0.1 username=me result=user:25
login ip=127.0.0.1 username=me2 result=user:26
login ip=127.0.0.1 username=me3 result=user:27
.login ip=127.0.0.1 username=me result=fail
login ip=127.0.0.1 username=notme result=fail
login ip=127.0.0.1 username=me result=user:29
login ip=127.0.0.1 username=me result=user:29
.login ip=127.0.0.1 username=me result=user:33
.login ip=127.0.0.1 username=me result=user:37
....login ip=127.0.0.1 username=me2 result=user:51
Elogin ip=127.0.0.1 username=me result=user:54
.login ip=127.0.0.1 username=me result=user:58
.login ip=127.0.0.1 username=me result=user:62
.login ip=127.0.0.1 username=me result=user:66
.login ip=127.0.0.1 username=me result=user:70
.
======================================================================
ERROR: test_settings_page (siteapp.tests.OrganizationSettingsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kaminski/src/govready/govready-q/siteapp/tests.py", line 1078, in test_settings_page
    response = self.client.get('/settings')
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 517, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 332, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 404, in generic
    return self.request(**r)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 485, in request
    raise exc_value
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/core/handlers/exception.py", line 35, in inner
    response = get_response(request)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/core/handlers/base.py", line 128, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/Users/kaminski/src/govready/govready-q/siteapp/views.py", line 1486, in organization_settings
    organization = Organization.objects.get(id=1)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/db/models/query.py", line 403, in get
    self.model._meta.object_name
siteapp.models.DoesNotExist: Organization matching query does not exist.

----------------------------------------------------------------------
Ran 37 tests in 318.024s

FAILED (errors=1)
Destroying test database for alias 'default'...
```

**Comment by gregelin** [comment]
After further research, the`test_settings_page` test page fails because of a brittle Organization look up in the database at the beginning of the view for the `/settings` page. The brittle Organization look up Django is running in `siteapp.views.organization_settings` assumes there will be an Organization with `id=1`. And Organization iwth `id=1` does not exist in the test database during at this stage of the test. The test Organization that is in the database has `id=2`.

This is why the test is reporting an "Error" instead of the a "Failed." The application code is crashing during the test and throwing the error `Organization matching query does not exist.`

The buggy code block was the result of a temporary fix in `0.9.0.dev` branch related to the removal of sub-domain based multi-tenancy.  The buggy code had a comment identifying the weakness:

```python
    # In 0.9.0 we only have 1 organization, so let's get that
    # TODO better setting of organization
    organization = Organization.objects.get(id=1)
```

We are going to fix this by having the application code return the first organization in the database. 

```python
    # In 0.9.0 we should only have 1 organization, so let's get that
    # If Instance has been upgraded to 0.9.0 from 0.8.6 there will be multiple organizations,
    #   but we are still going to be looking for the first organization created during install.
    #   This approach is more robust but still brittle. It is possible for an administrator to directly
    #   change the organization database tables resulting in a different organization being "first."
    #   A more robust approach might be adding a field to Organization model to identify the
    #   "main" organization and allow only one main organization to exist.
    # TODO better setting of organization
    organization = Organization.objects.first()
```

**Comment by gregelin** [comment]
Fixed in commit 2eeab8b9a7d245260d1bc026e809c07f5cd93983

**Comment by gregelin** [comment]
Fixed.
